Mathlib/Logic/Basic.lean

```diff
@@ -253,6 +253,14 @@ instance : Std.Commutative Xor' := ⟨xor_comm⟩
 
 theorem xor_not_not : Xor' (¬a) (¬b) ↔ Xor' a b := by simp [Xor', or_comm, and_comm]
 
+@[simp] theorem and_not_self_and_iff_and_not (a b : Prop) :  a ∧ (a → ¬b) ↔ a ∧ ¬b := by
+  rw [and_congr_right_iff]
+  exact imp_iff_right
+
+theorem xor_iff_or_and_not_and (a b : Prop) : Xor' a b ↔ (a ∨ b) ∧ (¬ (a ∧ b)) := by
+  rw [Xor', or_and_right, not_and, and_not_self_and_iff_and_not, imp_not_comm,
+    and_not_self_and_iff_and_not (b := a) (a := b)]
```

Comment:
```suggestion
theorem xor_iff_or_and_not_and (a b : Prop) : Xor' a b ↔ (a ∨ b) ∧ (¬ (a ∧ b)) := by
  simp_rw [Xor', or_and_right, not_and_or, and_or_left, and_not_self_iff, false_or, or_false]
```

Unless you have some other reason to include that other lemma of course.

Reply:
@ADedecker that doesn't appear to work - or at least depends on tactics / results not yet available at this point in the build.

Reply:
How about?
```
theorem xor_iff_or_and_not_and (a b : Prop) : Xor' a b ↔ (a ∨ b) ∧ (¬ (a ∧ b)) := by
  rw [Xor', or_and_right, not_and_or, and_or_left, and_not_self_iff, false_or,
    and_or_left, and_not_self_iff, or_false]
```

Reply:
TIL `simp_rw` is not imported by all of Mathlib! Sorry for the troubles.